linkouts/linkouts/__main__.py

```diff
@@ -0,0 +1,144 @@
+class EnvArg(argparse.Action):
+
+  def __init__(self, envvar, required=False, default=None, **kwargs):
+    if envvar in os.environ:
+      default = os.environ[envvar]
+    if required and default is not None:
+      required = False
+    super(EnvArg, self).__init__(default=default, required=required, **kwargs)
+
+  def __call__(self, parser, namespace, values, option_string=None):
+    setattr(namespace, self.dest, values)
+
+
+
+def parseArgs():
+
+  # create the parser
+  parser = argparse.ArgumentParser(
+    prog=linkouts.__name__,
+    description='A microservice that returns the hyperlink info objects corresponding to the given gene id.',
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+  parser.add_argument(
+    '--version',
+    action='version',
+    version=f'%(prog)s {linkouts.__version__}',
+  )
+
+  # logging args
+  loglevel_envvar = 'LOG_LEVEL'
+  parser.add_argument(
+    '--log-level',
+    dest='log_level',
+    action=EnvArg,
+    envvar=loglevel_envvar,
+    type=str,
+    choices=list(LOG_LEVELS.keys()),
+    default='WARNING',
+    help=('What level of events should be logged (can also be specified using '
+          f'the {loglevel_envvar} environment variable).'))
+  logfile_envvar = 'LOG_FILE'
+  parser.add_argument(
+    '--log-file',
+    dest='log_file',
+    action=EnvArg,
+    default=argparse.SUPPRESS,  # removes "(default: None)" from help text
+    envvar=logfile_envvar,
+    type=str,
+    help=('The file events should be logged in (can also be specified using '
+          f'the {logfile_envvar} environment variable).'))
+
+
+  # Async HTTP args
+  hhost_envvar = 'HTTP_HOST'
+  parser.add_argument('--hhost', action=EnvArg, envvar=hhost_envvar, type=str, default='127.0.0.1', help=f'The HTTP server host (can also be specified using the {hhost_envvar} environment variable).')
+  hport_envvar = 'HTTP_PORT'
+  parser.add_argument('--hport', action=EnvArg, envvar=hport_envvar, type=str, default='8880', help=f'The HTTP server port (can also be specified using the {hport_envvar} environment variable).')
+  lglob_root_envvar = ''
+  parser.add_argument('--lglob_root', action=EnvArg, envvar=lglob_root_envvar, type=str, default='/data', help=f'The root folder to be searched for linkouts.*.yml files containing linkout specifications')
+
+  return parser.parse_args()
+
+
+# the main coroutine that starts the various program tasks
+def main_coroutine(args):
+  tasks = []
+  handler = RequestHandler(args.lglob_root)
+  run_http_server(args.hhost, args.hport, handler)
+
+def main():
+
+  # parse the command line arguments / environment variables
+  args = parseArgs()
+
+  # setup logging
+  log_config = {
+    'format': '%(asctime)s,%(msecs)d %(levelname)s: %(message)s',
+    'datefmt': '%H:%M:%S',
+    'level': LOG_LEVELS[args.log_level],
+  }
+  if 'log_file' in args:
+    log_config['filename'] = args.log_file
+  logging.basicConfig(**log_config)
+
+  # initialize asyncio
+  #uvloop.install()
```

Comment:
This line should be removed; that uvloop function is deprecated so it won't even work in future releases.